### ISSUE TITLE
Fujifilm X-T4 compressed RAW not decoded properly (black screen)

### ISSUE THREAD

**Issue by nicholascw** [Beep6581/RawTherapee#5845]
<details>
  <summary>RawTherapee version</summary>

```
// Archlinux Repository community/rawtherapee version 1:5.8-1

Version: 5.8
Branch: releases
Commit: 9a9e0acbf
Commit date: 2020-02-04
Compiler: cc 9.2.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.24.2
Lensfun: V0.3.95.0
Build type: Release
Build flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=c++11  -Werror=unused-label -Werror=delete-incomplete -Wall -Wuninitialized -Wcast-qual -Wno-deprecated-declarations -Wno-unused-result -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ftree-vectorize
Link flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
OpenMP support: ON
MMAP support: ON
```
</details>

<details>
  <summary>System Version</summary>

```
Archlinux up-to-date
Linux 5.7.7-arch1-1 #1 SMP PREEMPT Wed, 01 Jul 2020 14:53:16 +0000 x86_64
```

</details>

### Symptoms

Throughout test around different settings I found:
 - [ ] The Compressed RAW totally blacked out. (should be simply not supported)
 - [x] (fixed, confirmed with 8af287aa8c9659ae2e0f2fff31f8de8c854f63d7) Only a center portion of the image are shown for Continuous High with the resolution been cropped down to 6262x994 rather than 6262x4126, and the file size shown in RawTherapee reduced significantly, too. (See screenshot below)
 - [x] (fixed, confirmed with 8af287aa8c9659ae2e0f2fff31f8de8c854f63d7) All with black stripes on the right and bottom. (Probably caused by the new IBIS? But turning off the IBIS has the same black strip on the edge.)

![图片](https://user-images.githubusercontent.com/7414442/87242727-d6b76080-c3f4-11ea-8280-be8199f59ac2.png)

[Here (mega.nz-shared)](https://mega.nz/file/YHB2HYJC#VPkjZh1f3dcRaAY03wjAegKjKuj2R5oCFsXANThdBcE) are my RAW files (together with .pp3 metadata) packed with **manifest.txt** explaining what parameters each of the files are shooting with. I have tried to iterated through different settings to reproduce the bug. Also to be noted, these issues are persistent for all 1.0 / 1.01 / 1.02 firmware. [Folder link is here](https://mega.nz/folder/BHgjWYwL#5Y8iodQisgszbHnNN9KbEg) for easy access and smaller the download size.

<details>
  <summary>Contents of the ZIP archive</summary>

```
Filename            Behavior        Settings
_DSF1443.RAF        Normal          Uncompressed / AdobeRGB / Single
_DSF1444.RAF        Normal          Lossless     / AdobeRGB / Single
_DSF1445.RAF        Black           Compressed   / AdobeRGB / Single
DSCF1446.RAF        Normal          Uncompressed / sRGB     / Single
DSCF1448.RAF        Normal          Lossless     / sRGB     / Single
DSCF1449.RAF        Black           Compressed   / sRGB     / Single
DSCF1450.RAF        Black           Compressed   / sRGB     / HDR
DSCF1451.RAF        Normal          Lossless     / sRGB     / HDR
DSCF1452.RAF        Normal          Uncompressed / sRGB     / HDR
DSCF1453.RAF        Cropped         Uncompressed / sRGB     / Continuous High
DSCF1455.RAF        Cropped         Lossless     / sRGB     / Continuous High
DSCF1456.RAF        Black           Compressed   / sRGB     / Continuous High
```

</details>

Let me know if any further tests are needed. I'll be willing to compile the master branch or attach a gdb debugger (though it seems to be a parsing error, should be easily reproduced rather than a build/machine specific), or more RAW files with different settings.

**Comment by Thanatomanic** [comment]
@nicholascw Thank you for this extensive and well-written report. We had a similar issue [here](https://github.com/Beep6581/RawTherapee/issues/5709) regarding the strange crop, so I guess a fix should be relatively easy.

**Comment by heckflosse** [comment]
Iirc that's already fixed in 5.8 dev

**Comment by Thanatomanic** [comment]
@heckflosse @nicholascw Indeed, the crop was already fixed: https://github.com/Beep6581/RawTherapee/commit/ca605eece6cb799c0e349dd6303c55300ce7e69c

The compressed format doesn't work though.

**Comment by heckflosse** [comment]
> The compressed format doesn't work though.

Confirmed. I will have a look if the compressed format works in ART. If it does, we may borrow some code.

**Comment by nicholascw** [comment]
@Thanatomanic Thanks for the quick update. I've confirmed that those two are fixed with the latest master branch and updated the issue to align with the title. Thanks again for nice work done by all of you!

**Comment by nicholascw** [comment]
> Confirmed. I will have a look if the compressed format works in ART. If it does, we may borrow some code.

@heckflosse Oops, just tried on Archlinux with AUR pkgbuild script **art-rawconverter-git**, which pulls their master branch as build source, it seems not working properly as well.

RawTherapee:
![图片](https://user-images.githubusercontent.com/7414442/87244236-7aa70900-c401-11ea-9f2a-0f676ea2afd6.png)

ART:
![图片](https://user-images.githubusercontent.com/7414442/87244278-ad510180-c401-11ea-9550-279cd4ecf7bb.png)

**Comment by oliver124810** [comment]
I confirm that the "Continuous High" bug is fixed on master but not in 5.8.
